test/integration/filters/test_listener_filter.h

```diff
@@ -0,0 +1,31 @@
+#include "envoy/registry/registry.h"
+#include "envoy/server/filter_config.h"
+
+namespace Envoy {
+/**
+ * Test listener filter which sets the ALPN to a manually configured string.
+ */
+class TestListenerFilter : public Network::ListenerFilter {
+public:
+  TestListenerFilter() = default;
+
+  // Network::ListenerFilter
+  Network::FilterStatus onAccept(Network::ListenerFilterCallbacks& cb) override {
+    absl::MutexLock m(&alpn_lock_);
+    ASSERT(!alpn_.empty());
+    cb.socket().setRequestedApplicationProtocols({alpn_});
+    alpn_.clear();
+    return Network::FilterStatus::Continue;
+  }
+
```

Comment:
We got a new method for ListenerFilter

https://github.com/envoyproxy/envoy/blob/08c2e5a71b7ca5638b1a3d6ce062bfd8c5e87b28/envoy/network/filter.h#L341

Just need a empty implement of onData